mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/mps/factories/SNodeFactory.kt

```diff
@@ -136,10 +136,13 @@ class SNodeFactory(
             val sourceNodeId = iNode.nodeIdAsLong()
             val source = nodeMap.getNode(sourceNodeId)!!
 
-            val sReferenceLink = (it.first as MPSReferenceLink).link
-            val reference = SReferenceLinkAdapter2(sReferenceLink.id, sReferenceLink.name)
-            val targetNodeId = it.second.nodeIdAsLong()
+            val reference = when (it.first) {
+                is MPSReferenceLink -> (it.first as MPSReferenceLink).link
+                is ReferenceLinkFromName -> source.concept.referenceLinks.first { refLink -> refLink.name == it.first.getSimpleName() }
```

Comment:
A comment on why `ReferenceLinkFromName` can happen.
Or even better a test...

Reply:
In general, I'd like to have tests for all new features except Swing-UI code (that should be kept minimal)... Not sure how realistic it is.

Reply:
Due to time pressure, I would prefer to skip this aspect and do it when we get there: https://issues.modelix.org/issue/MODELIX-536

Reply:
> A comment on why ReferenceLinkFromName can happen.

I'm not sure how that can happen. I just looked for all classes that implement the IReferenceLink interface. However, `GeneratedReferenceLink` is not used here, so I'll add that to make the implementation complete.

(The only missing class is SimpleReferenceLink, but that is only used in our code-based metamodel of the RepositoryConcepts and so.)

Reply:
☝️  is fixed in b79fb24de7633fd5813ba2e45c2b89c706344a88

Reply:
```kt
      val reference = when (val referenceLink = it.first) {
                is MPSReferenceLink -> referenceLink.link
                // `referenceLink` is `IReferenceLink`. `IReferenceLink.getSimpleName()` always exists.
                else -> source.concept.referenceLinks.first { refLink -> refLink.name == referenceLink.getSimpleName() }
            }
 ```

You can just use `IReferenceLink.getSimpleName()` directly without type checks.

Reply:
fixed in 1fdcf251089c424d0847244bb84abde07d71c7a4